lib/webauthn/base_validation_service.rb

```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
 module DiscourseWebauthn
-  class SecurityKeyBaseValidationService
-    def initialize(current_user, params, challenge_params)
+  class BaseValidationService
+    def initialize(current_user, params, options = {})
```

Comment:
```suggestion
    def initialize(current_user, params, factor_type:, session:)
```

Small suggestion but we can use keyword arguments instead which ensures the right keys are always used. If we use a hash, there is a chance that a typo may end up causing `@factor_type` or `@session` to not be set.